### PR TITLE
Add possibility to show facets with 0 matching documents

### PR DESCRIPTION
**Pull request by bidoubiwa**
fixes: #599 

Some improvements were done on the readme as well

Screenshot 2022-03-17 at 18 16 54

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,18 +89,65 @@ const searchClient = instantMeiliSearch(
...
+    // ...
   }
 )
 ```
 
-- `placeholderSearch` (`true` by default). Displays documents even when the query is empty.
+### Placeholder Search
+
+Placeholders search means showing results even when the search query is empty. By default it is `true`.
+If set to `false`, only when a request is made with at least one character, results will appear.
```
Comment (alallema) [inline README.md:100]: It is maybe came from me but I find this sentence hard to understand

File: README.md [hunk 2]
```diff
@@ -85,18 +89,65 @@ const searchClient = instantMeiliSearch(
...
-  ⚠️ Meilisearch is not designed for pagination and this can lead to performances issues, so the usage of the pagination widget is not encouraged. However, the `paginationTotalHits` parameter lets you implement this pagination with less performance issue as possible: depending on your dataset (the size of each document and the number of documents) you might decrease the value of `paginationTotalHits`.<br>
-  More information about Meilisearch and the pagination [here](https://github.com/meilisearch/documentation/issues/561).
-- `primaryKey` (`undefined` by default): Specify the field in your documents containing the [unique identifier](https://docs.meilisearch.com/learn/core_concepts/documents.html#primary-field). By adding this option, we avoid instantSearch errors that are thrown in the browser console. In `React` particularly, this option removes the `Each child in a list should have a unique "key" prop` error.
+`keepZeroFacets` set to true keeps the facets even when they have 0 matching documents (default `false`).
```
Comment (alallema) [inline README.md:131]:
```suggestion
`keepZeroFacets` set to `true` keeps the facets even when they have 0 matching documents (default `false`).
```

File: README.md (hunk 1 above)
Comment (alallema) [reply README.md:100]:
```
If set to `false`, results will appear only when a request is made with at least one character.
```

**Review by alallema — COMMENTED** [review]
Well done! I had just few comments

File: README.md (hunk 1 above)
Comment (bidoubiwa) [reply README.md:100]: > When placeholder search is set to `false`, no results appears when searching on no characters. For example, if the query is `""` no results appear.

Comment (alallema) [reply README.md:100]: Perfect

**Review by alallema — APPROVED** [review]
LGTM! 💪

**Comment by bidoubiwa** [comment]
bors merge

**Comment by ar-siddiqui** [comment]
Is this in production yet? This is still a pain point for us.

**Comment by bidoubiwa** [comment]
It is in production, could you open an issue with your specific setup that does not work?

**Pull request merged**